File: benchmark/java/src/main/java/com/benchmark/services/DeserializationService.java
package com.benchmark.services;

import java.io.ByteArrayInputStream;
import java.io.ObjectInputStream;

public class DeserializationService {
    public void deserializeObject(byte[] data) {
        try {
            ByteArrayInputStream bais = new ByteArrayInputStream(data);
            // Insecure Deserialization [CWE-502]
            ObjectInputStream ois = new ObjectInputStream(bais);
            Object obj = ois.readObject();
            ois.close();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: benchmark/csharp/Services/DeserializationService.cs
using System.IO;
using System.Runtime.Serialization.Formatters.Binary;

namespace Benchmark.CSharp.Services
{
    public class DeserializationService
    {
        public object DeserializeData(byte[] data)
        {
            // Insecure Deserialization [CWE-502]
            BinaryFormatter formatter = new BinaryFormatter();
            using (MemoryStream ms = new MemoryStream(data))
            {
                return formatter.Deserialize(ms);
            }
        }
    }
}
